go-controller/pkg/ovn/external_ids_syncer/port_group/port_group_sync.go

```diff
@@ -0,0 +1,230 @@
+package port_group
+
+import (
+	"fmt"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"strings"
+
+	libovsdbclient "github.com/ovn-org/libovsdb/client"
+	libovsdb "github.com/ovn-org/libovsdb/ovsdb"
+	libovsdbops "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdb/ops"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	"k8s.io/klog/v2"
+)
+
+const (
+	// port groups suffixes
+	// ingressDefaultDenySuffix is the suffix used when creating the ingress port group for a namespace
+	ingressDefaultDenySuffix = "ingressDefaultDeny"
+	// egressDefaultDenySuffix is the suffix used when creating the ingress port group for a namespace
+	egressDefaultDenySuffix      = "egressDefaultDeny"
+	defaultNetworkControllerName = "default-network-controller"
+)
+
+type updatePortGroupInfo struct {
+	acls  []*nbdb.ACL
+	oldPG *nbdb.PortGroup
+	newPG *nbdb.PortGroup
+}
+
+type PortGroupSyncer struct {
+	nbClient libovsdbclient.Client
+	// txnBatchSize is used to control how many port groups will be updated with 1 db transaction.
+	// Set to 0 to disable batching
+	txnBatchSize int
+}
+
+func getControllerName(networkExternalID string) string {
+	if networkExternalID == "" {
+		return defaultNetworkControllerName
+	}
+	return networkExternalID + "-network-controller"
+}
+
+// NewPortGroupSyncer creates a PortGroupSyncer that will sync port groups without the new ExternalIDs for all
+// controllers. Controller name will be defined based on getControllerName() function.
+func NewPortGroupSyncer(nbClient libovsdbclient.Client) *PortGroupSyncer {
+	return &PortGroupSyncer{
+		nbClient:     nbClient,
+		txnBatchSize: 50,
+	}
+}
+
+func getPortGroupMulticastDbIDs(namespace, networkExternalID string) *libovsdbops.DbObjectIDs {
+	controllerName := getControllerName(networkExternalID)
```

Comment:
heh I love this grouped set of `getdbIDs` functions, I wish we could just put that into libovsdb.utils somewhere without needing to do one version of these in the main ovn pkg and another version again in the external_ids_syncer pkg...

Reply:
unfortunately we can't do that, because syncer package is only used to translate before-dbids objects to after-dbids. If we ever change new ids, the new sync will be done in another place (starting from previous db IDs version), and this file should never change.